### ISSUE TITLE
Support Health Connect background fetch

### ISSUE THREAD

**Issue by armanso** [cph-cachet/flutter-plugins#1058]
HC introduced background fetch in [Version 1.1.0-alpha09](https://developer.android.com/jetpack/androidx/releases/health-connect#1.1.0-alpha09). According to the issue tracker I have, this feature is in production for the latest Android 14 version and above. Support for earlier versions will come later. Since it’s possible to check if this feature is available using a [feature check](https://developer.android.com/health-and-fitness/guides/health-connect/develop/feature-availability), I would like to integrate this into a Flutter plugin as well.

I would be happy to take the lead on this and start the implementation if it’s not already on the roadmap.

You can find the documentation regarding background fetch [here](https://developer.android.com/health-and-fitness/guides/health-connect/develop/read-data#background-read-example).

**Comment by arsalancy** [comment]
@armanso Hi, any update on this?

**Comment by giacomoto** [comment]
Is it in progress?

**Comment by armanso** [comment]
Hey everyone!
Sorry for the late reply — life got a bit busy. I've made some progress on this and I'm happy to continue working on it.

That said, I'm not sure if the team behind this plugin is also working on this feature. Since I couldn’t find a CONTRIBUTING.md file or any guidelines on who to contact for approvals, I’m assuming @iarata is the right person to check with regarding commits and merges.

**Comment by dennys** [comment]
I can use health plugin to get data in app. But when I use workmanager to get health data, it always get 0 for steps. Is this the root cause?

**Comment by armanso** [comment]
@dennys that's correct! Health Connect function can't be called directly from a worker. It requires additional permissions and a specific setup in your code to work properly.

https://developer.android.com/health-and-fitness/guides/health-connect/develop/read-data#background-read-example

**Comment by dennys** [comment]
I try to find an alternative, but it seems I need to use Google Fit, or calculate the step from last reboot and calculate it by myself. Do you have any suggestion?

**Comment by armanso** [comment]
@dennys Not sure what your use case is, but if you only need step tracking in the background, you can always use the step sensor + foreground service. Just keep in mind that you'll need to handle data storage yourself.

**Comment by armanso** [comment]
Thanks, @dennys! That's great news that they're working on this feature.
